### PR TITLE
Code commit for poc

### PR DESCRIPTION
**Pull request by srajences**

### CODE REVIEW

File: cypress/utils/navigate.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export class Navigate {
+/*This function handles the webPopup to access the location. It will make it false*/
+    static HandleLocationPopup(){
+        Cypress.on('window:confirm',(message)=>{
+            if(message.includes('Know your location')){
+                return false;
+            }
+        })
```
Comment (ajinkyaces) [inline cypress/utils/navigate.ts:8]: What is the use of this function ?

File: cypress/pageFunction/HomePageFunctions.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { HomePageLocators } from '../locators/HomePageLocators';
+import { PageTextValues } from '../pageEnums/PageTextValues';
+/*This function basically performs the Homepage component validations*/
+export class HomePageFunctions{
+    static validateTitle(expectedTitle:string){
+        cy.title().should('eq',expectedTitle);
+    }
```
Comment (ajinkyaces) [inline cypress/pageFunction/HomePageFunctions.ts:7]: Do we need a function for this ? Can be used directly !

File: cypress/pageFunction/HomePageFunctions.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        cy.get(HomePageLocators.kingDeals).should('be.visible').invoke('text').then((text)=>{
+            expect(text).to.eq(PageTextValues.KingDealsText)
+        });
+        cy.get(HomePageLocators.loginComponent).should('be.visible').invoke('text').then((text)=>{
+            expect(text).to.eq(PageTextValues.LoginText)
+        });
+        cy.get(HomePageLocators.cartComponent).should('be.visible').invoke('text').then((text)=>{
+            expect(text).to.eq(PageTextValues.CartText)
+        });
+
```
Comment (ajinkyaces) [inline cypress/pageFunction/HomePageFunctions.ts:14]: no need of promise can be used by should post invoke text

File: cypress/pageFunction/AddCartFunctions.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import cypress from 'cypress';
+import { AddCartPageLocators } from '../locators/AddCartPageLocators';
+import { PageTextValues } from '../pageEnums/PageTextValues';
+import { Runnable } from 'mocha';
+export class AddCartFunctions{
+    static selectLocation(){
+        cy.fixture('requiredData').then((data)=>{
+            const location = data.Location;
```
Comment (ajinkyaces) [inline cypress/pageFunction/AddCartFunctions.ts]: Do we need fixture data in this page function class ?

File: cypress/locators/HomePageLocators.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export class HomePageLocators{
+    /* This file contains the require locators for Home Page Validation*/
+    static dineInTakeway = 'div.toggle.delivery div.u_cursor_pointer.toggle__left '
+    static nearbuyStores = 'ul.nav-menu__list.delivery li:nth-child(1)'
+    static kingDeals = 'ul.nav-menu__list.delivery li:nth-child(2)'
+    static loginComponent = 'ul.nav-menu__list.delivery li:nth-child(3)'
+    static cartComponent = 'ul.nav-menu__list.delivery li:nth-child(4)'
```
Comment (ajinkyaces) [inline cypress/locators/HomePageLocators.ts]: These values needs to be constants

File: cypress/locators/AddCartPageLocators.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    static inputLocationField = 'input[type="text"]'
+    static okayButton = 'span.button__content:contains("okay")'
+    static confirmButton = 'span.button__content:contains("Confirm Location")'
+    static menuSelect = '.menu-item__name:contains("Whopper")'
+    static productPageName = '.product-page__name'
+    static toggleButton = '.toggle__custom.toggle__custom_uncheck'
+    static inputBar = '.search-autocomplete__search-input.location-search-input'
+    static addToCart = 'div.product-page__item:nth-child(1) div.card-gen__addtocart'
+    static continueButton = 'div.cardBlock button:nth-child(2)'
+}
```
Comment (ajinkyaces) [inline cypress/locators/AddCartPageLocators.ts]: Constant

File: cypress/e2e/HomePage.cy.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import {navigate} from '../utils/navigate';
+import { HomePageFunctions } from '../pageFunction/HomePageFunctions';
+import { PageTextValues } from '../pageEnums/PageTextValues';
+/* Author: Srajen Singh*/
+beforeEach(()=>{
+    
+    cy.fixture('requiredData').then((data)=>{
+        const url = data.url;
+        navigate(url);
```
Comment (ajinkyaces) [inline cypress/e2e/HomePage.cy.ts]: Definition must be outside the before-Each block

File: cypress/e2e/AddCartPage.cy.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import { AddCartFunctions } from '../pageFunction/AddCartFunctions';
+import {navigate} from '../utils/navigate';
+/* Author: Srajen Singh*/
+beforeEach(()=>{
+    
+    cy.fixture('requiredData').then((data)=>{
+        const url = data.url;
+        navigate(url);
```
Comment (ajinkyaces) [inline cypress/e2e/AddCartPage.cy.ts]: Url definition should be outside the before each block

**Review by ajinkyaces — CHANGES_REQUESTED** [review]

File: cypress/utils/navigate.ts (hunk 1 above)
Comment (srajences) [reply cypress/utils/navigate.ts:8]: @ajinkyaces  use of this function is for handling popup regarding granting location access. This function is used for cancelling the popup

File: cypress/pageFunction/HomePageFunctions.ts (hunk 3 above)
Comment (srajences) [reply cypress/pageFunction/HomePageFunctions.ts:14]: @ajinkyaces sure I will change it

File: cypress/pageFunction/HomePageFunctions.ts (hunk 2 above)
Comment (srajences) [reply cypress/pageFunction/HomePageFunctions.ts:7]: @ajinkyaces yeah we don't need function for this but for avoiding assertion in test file that's why I created a function

File: cypress/e2e/AddCartPage.cy.ts (hunk 8 above)
Comment (srajences) [reply cypress/e2e/AddCartPage.cy.ts]: @ajinkyaces sure I will make changes in next pr

File: cypress/locators/AddCartPageLocators.ts (hunk 6 above)
Comment (srajences) [reply cypress/locators/AddCartPageLocators.ts]: will address this in next pr

File: cypress/pageFunction/AddCartFunctions.ts (hunk 4 above)
Comment (srajences) [reply cypress/pageFunction/AddCartFunctions.ts]: @ajinkyaces yeah for reading location value for json but I will change it. I will try to pass it from test class

**Comment by ajinkyaces** [comment]
Avoid committing the report folder, use the gitignore file for the same

**Review by ajinkyaces — APPROVED** [review]

**Pull request merged**